Mathlib/LinearAlgebra/TensorProduct/Quotient.lean

```diff
@@ -0,0 +1,241 @@
+-/
+noncomputable def quotientTensorQuotientEquiv (m : Submodule R M) (n : Submodule R N) :
+    (M ⧸ (m : Submodule R M)) ⊗[R] (N ⧸ (n : Submodule R N)) ≃ₗ[R]
+    (M ⊗[R] N) ⧸
+      (LinearMap.range (map m.subtype LinearMap.id) ⊔
+        LinearMap.range (map LinearMap.id n.subtype)) :=
+  LinearEquiv.ofLinear
+    (lift <| Submodule.liftQ _ (LinearMap.flip <| Submodule.liftQ _
+      ((mk R (M := M) (N := N)).flip.compr₂ (Submodule.mkQ _)) fun x hx => by
+      ext y
+      simp only [LinearMap.compr₂_apply, LinearMap.flip_apply, mk_apply, Submodule.mkQ_apply,
+        LinearMap.zero_apply, Submodule.Quotient.mk_eq_zero]
+      exact Submodule.mem_sup_right ⟨y ⊗ₜ ⟨x, hx⟩, rfl⟩) fun x hx => by
+      ext y
+      simp only [LinearMap.coe_comp, Function.comp_apply, Submodule.mkQ_apply, LinearMap.flip_apply,
+        Submodule.liftQ_apply, LinearMap.compr₂_apply, mk_apply, LinearMap.zero_comp,
+        LinearMap.zero_apply, Submodule.Quotient.mk_eq_zero]
+      exact Submodule.mem_sup_left ⟨⟨x, hx⟩ ⊗ₜ y, rfl⟩)
+    (Submodule.liftQ _ (map (Submodule.mkQ _) (Submodule.mkQ _)) fun x hx => by
+      rw [Submodule.mem_sup] at hx
+      rcases hx with ⟨_, ⟨a, rfl⟩, _, ⟨b, rfl⟩, rfl⟩
+      simp only [LinearMap.mem_ker, map_add]
+      set f := (map m.mkQ n.mkQ) ∘ₗ (map m.subtype LinearMap.id)
+      set g := (map m.mkQ n.mkQ) ∘ₗ (map LinearMap.id n.subtype)
+      have eq : LinearMap.coprod f g = 0 := by
+        ext x y
+        · simp [f, Submodule.Quotient.mk_eq_zero _ |>.2 x.2]
+        · simp [g, Submodule.Quotient.mk_eq_zero _ |>.2 y.2]
+      exact congr($eq (a, b)))
+    (by ext; simp) (by ext; simp)
+
+@[simp]
+lemma quotientTensorQuotientEquiv_apply_tmul_mk_tmul_mk
+    (m : Submodule R M) (n : Submodule R N) (x : M) (y : N) :
+    quotientTensorQuotientEquiv m n
+      (Submodule.Quotient.mk x ⊗ₜ[R] Submodule.Quotient.mk y) =
+      Submodule.Quotient.mk (x ⊗ₜ y) := rfl
+
+@[simp]
+lemma quotientTensorQuotientEquiv_symm_apply_mk_tmul
+    (m : Submodule R M) (n : Submodule R N) (x : M) (y : N) :
+    (quotientTensorQuotientEquiv m n).symm (Submodule.Quotient.mk (x ⊗ₜ y)) =
+      Submodule.Quotient.mk x ⊗ₜ[R] Submodule.Quotient.mk y := rfl
+
+variable (N) in
+/--
+Let `M, N` be `R`-modules, `m ≤ M` be an `R`-submodule. Then we have a linear isomorphism between
+tensor products of the quotient and the quotient of the tensor product:
+`(M ⧸ m) ⊗[R] N ≃ₗ[R] (M ⊗[R] N) ⧸ (m ⊗ N)`.
+-/
+noncomputable def quotientTensorEquiv (m : Submodule R M) :
+    (M ⧸ (m : Submodule R M)) ⊗[R] N ≃ₗ[R]
+    (M ⊗[R] N) ⧸ (LinearMap.range (map m.subtype (LinearMap.id : N →ₗ[R] N))) :=
+  congr (LinearEquiv.refl _ _) ((Submodule.quotEquivOfEqBot _ rfl).symm) ≪≫ₗ
+  quotientTensorQuotientEquiv (N := N) m ⊥ ≪≫ₗ
+  Submodule.Quotient.equiv _ _ (LinearEquiv.refl _ _) (by
+    simp only [Submodule.map_sup]
+    erw [Submodule.map_id, Submodule.map_id]
+    simp only [sup_eq_left]
+    rw [map_range_eq_span_tmul, map_range_eq_span_tmul]
+    aesop)
```

Comment:
You could say `quotientTensorEquiv` is a special of `rTensor.equiv`. But my definition is a special case of 
`(M / m) ⊗ (N / n) ≃ (M ⊗ N) / (m ⊗ N sup M ⊗ n)` i.e. [TensorProduct.quotientTensorQuotientEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/TensorProduct/Quotient.html#TensorProduct.quotientTensorQuotientEquiv). In this file [TensorProduct.quotientTensorQuotientEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/TensorProduct/Quotient.html#TensorProduct.quotientTensorQuotientEquiv) is the most general form and the rest is derived from this

Reply:
`(rTensor.equiv N (LinearMap.exact_subtype_mkQ _) m.mkQ_surjective).symm` works but then the first `rfl` below fails, so your construction is indeed nicer definitionally. This is because `rTensor.equiv` needs to choose a right inverse to the surjective map from the second module (M here) to the third module (M/m here). Only when the third module is a quotient is it possible to use `Quotient.lift` (as you do in this PR).